UX: chat message thread indicator v2 design & refactor

chapoi opened the pull request:
This PR contains:
* gridified the thread message indicator, alleviating some problems with positioning and overflow
* participant avatars will overlap/smush on smaller size and mobile
* the excerpt went from 3 > 2 lines of wrapping on smaller size, still 1 line on large size
* dropped the copy of "last reply"
* fixed wrong line height
* moved the "x replies" over to the right near the participants, as that makes more sense
* using a bubble to indicate other participants, instead of copy

### desktop & drawer
![image](https://github.com/discourse/discourse/assets/101828855/3f325fbd-fcc0-40ba-838f-a287228c3001)

<img width="438" alt="image" src="https://github.com/discourse/discourse/assets/101828855/d497e917-babc-46be-996e-e7ceec02e496">


<img width="589" alt="image" src="https://github.com/discourse/discourse/assets/101828855/e365c810-66b0-4648-8780-dee6a5987e23">

### mobile
identical to drawer/small screen

And before anyone asks: it IS centered, so misalignment you may see is an optical illusion.
![image](https://github.com/discourse/discourse/assets/101828855/0f7889c5-1f9e-4ee0-a254-134964d0133d)

And as a footnote: this PR introduces the `@container` query, which is experimental. Nothing will break when it's being viewed in a not-supported browser, but it will be less elegant.

jjaffeux commented on plugins/chat/assets/javascripts/discourse/components/chat-message-thread-indicator.hbs:
```diff
@@ -8,50 +8,26 @@
+      {{format-date @message.thread.preview.lastReplyCreatedAt leaveAgo="true"}}
+    </span>
+  </div>
+  {{! <div class="chat-message-thread-indicator__participants"> }}
```
did you mean to remove this?

chapoi replied:
Remove what?

jjaffeux replied:
The comment? `{{! <div class="chat-message-thread-indicator__participants"> }}` ?

jjaffeux commented:
On specific width of drawer I can end up in this case:

![Screenshot 2023-06-21 at 16 48 13](https://github.com/discourse/discourse/assets/339945/711f1dd6-9bc0-4fa7-8c92-7ebd4e7d77fb)

jjaffeux commented:
Number and avatar don't have same size:

![Screenshot 2023-06-21 at 16 51 12](https://github.com/discourse/discourse/assets/339945/a77eb054-2e3a-4746-ae17-c6994da65de6)
![Screenshot 2023-06-21 at 16 51 07](https://github.com/discourse/discourse/assets/339945/8f329405-8011-42ee-878a-b8ff9f41068d)

jjaffeux commented:
![Screenshot 2023-06-21 at 16 51 58](https://github.com/discourse/discourse/assets/339945/95bf1161-7feb-4add-92b9-32142bbfc1a7)

Shouldn't we try to align left of the thread indicator with the reaction/message above?

jjaffeux commented:
(on mobile at least) Alignment feels off to me here:

![Screenshot 2023-06-21 at 16 54 23](https://github.com/discourse/discourse/assets/339945/8863175e-8ddf-4be1-ae11-4c04f0e66eb6)


Also I wonder if dont want just a little bit more spacing between text of the message and above content.

jjaffeux commented:
@jordanvidrine could you review this please? I was making feedback, but I think it would be better if another designer was doing it.

chapoi commented:
> ![Screenshot 2023-06-21 at 16 49 17](https://user-images.githubusercontent.com/339945/247591859-6213f564-2914-4892-a7ae-5ddd23ac1900.png)
> 
> I guess it's related to your is-online question. Avatar is not left aligned here.

It is

chapoi commented:
> ![Screenshot 2023-06-21 at 16 49 17](https://user-images.githubusercontent.com/339945/247591859-6213f564-2914-4892-a7ae-5ddd23ac1900.png)
> 
> I guess it's related to your is-online question. Avatar is not left aligned here.

It is though. But visually, because of the transparent border it looks 1px misaligned. If we decide to keep is-online, I'll go add in a negative margin.

chapoi commented:
> ![Screenshot 2023-06-21 at 16 51 58](https://user-images.githubusercontent.com/339945/247592969-95bf1161-7feb-4add-92b9-32142bbfc1a7.png)
> 
> Shouldn't we try to align left of the thread indicator with the reaction/message above?

No? Because then you don't have any padding on the side…?

chapoi commented:
> (on mobile at least) Alignment feels off to me here, I know it's not supposed to be related or on the same line, but the fact that's it "almost" on the same line makes it look off IMO:
> 
> ![Screenshot 2023-06-21 at 16 54 23](https://user-images.githubusercontent.com/339945/247593888-8863175e-8ddf-4be1-ae11-4c04f0e66eb6.png)
> 
> Also I wonder if we dont want just a little bit more spacing between text of the message and above content.

Yeah I considered adding in more space, but to keep it optimised I didn't. Because it's only the avatar that's close, if I add more space, the text I felt it looked wrong too between username and text.

I disagree with the alignment being a problem. Like you said, it's a different part of the element, it shouldn't align.

jjaffeux commented:
> No? Because then you don't have any padding on the side…?

I guess I would agree if left avtar side was aligned with above message:

![Screenshot 2023-06-21 at 18 28 03](https://github.com/discourse/discourse/assets/339945/48fd8165-e85d-4d41-af72-cd47d8e8d956)

Something like that?

chapoi commented:
> On specific width of drawer I can end up in this case:
> 
> ![Screenshot 2023-06-21 at 16 48 13](https://user-images.githubusercontent.com/339945/247591531-711f1dd6-9bc0-4fa7-8c92-7ebd4e7d77fb.png)

fixed


> (on mobile at least) Alignment feels off to me here, I know it's not supposed to be related or on the same line, but the fact that's it "almost" on the same line makes it look off IMO:
> 
> ![Screenshot 2023-06-21 at 16 54 23](https://user-images.githubusercontent.com/339945/247593888-8863175e-8ddf-4be1-ae11-4c04f0e66eb6.png)

added space




> ![Screenshot 2023-06-21 at 18 28 03](https://user-images.githubusercontent.com/339945/247624141-48fd8165-e85d-4d41-af72-cd47d8e8d956.png)
> 
> Something like that? Anyways, just feedback, do what you think is best.

This Im deciding against. Aligning the block is more important, that's also how we do it for oneboxes and blockquotes.

jjaffeux commented:
> This Im deciding against. Aligning the block is more important, that's also how we do it for oneboxes and blockquotes.

Bookmarked to complain abut this again in 6 months 🔖

jordanvidrine commented:
I'll be taking a look at these today during travel.

Here are the first two things I see from the OP. Just want to make sure @chapoi that you updated the OP images?

The #1 here is not aligned with letters
<img width="300" alt="image" src="https://github.com/discourse/discourse/assets/30537603/42882167-720e-4e17-b3db-57b0be03d289">

What is the reason the furthest right avatar is larger than the rest?
<img width="300" alt="image" src="https://github.com/discourse/discourse/assets/30537603/12cc632b-0b6b-4cfb-9964-042f59580491">

I will have more when I get some more time to look.

chapoi commented:
> I'll be taking a look at these today during travel.
> 
> Here are the first two things I see from the OP. Just want to make sure @chapoi that you updated the OP images?
> 
> The #1 here is not aligned with letters <img alt="image" width="300" src="https://user-images.githubusercontent.com/30537603/248265703-42882167-720e-4e17-b3db-57b0be03d289.png">
> 
> What is the reason the furthest right avatar is larger than the rest? <img alt="image" width="300" src="https://user-images.githubusercontent.com/30537603/248265925-12cc632b-0b6b-4cfb-9964-042f59580491.png">
> 
> I will have more when I get some more time to look.

Yup, those mistakes had been pointed out and should be fixed. Ill update the images

jjaffeux commented:
@jordanvidrine I merged it because we don't want to sit on these improvements for too long, but feel free to make more comments we will do follow-ups
